pkg/kp/kv.go

```diff
@@ -110,7 +91,7 @@ type consulStore struct {
 	podStore podstore.Store
 }
 
-func NewConsulStore(client consulutil.ConsulClient) Store {
+func NewConsulStore(client consulutil.ConsulClient) *consulStore {
```

Comment:
Is there any value in using an interface vs concrete type here? I feel that the philosophical answer is "yes, of course, of course. hmm." What was your experience in interacting with the callsites of this method? Based on what I've seen so far, it seems like we were leaning of type inference anyway so it's probably a moot point.

… time passes …

To wit, golint decrees: `    94  53 warning         exported func NewConsulStore returns unexported type *kp.consulStore, which can be annoying to use (go-golint)

… still more time passes … 

I'm now concluding that this needs to be made public or into an interface. WDYT?